api/go.mod

```diff
@@ -14,6 +14,7 @@ require (
 	github.com/hashicorp/serf v0.10.1
 	github.com/mitchellh/mapstructure v1.4.1
 	github.com/stretchr/testify v1.7.0
+	golang.org/x/exp v0.0.0-20230321023759-10a507213a29
```

Comment:
so running `go mod tidy` here to get exp slices package caused a cascading need of changes to other `go.mod` files that import this module which is why this change seems larger than it is